src/System.IO/tests/BinaryReader/BinaryReaderTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public class BinaryReaderTests
+    {
+        protected virtual Stream CreateStream()
+        {
+            return new MemoryStream();
+        }
+
+        [Fact]
+        public void BinaryReader_DisposeTests()
+        {
+            // Disposing multiple times should not throw an exception
+            using (Stream memStream = CreateStream())
+            using (BinaryReader binaryReader = new BinaryReader(memStream))
+            {
+                binaryReader.Dispose();
+                binaryReader.Dispose();
+                binaryReader.Dispose();
+            }
+        }
+
+        [Fact]
+        public void BinaryReader_CloseTests()
+        {
+            // Closing multiple times should not throw an exception
+            using (Stream memStream = CreateStream())
+            using (BinaryReader binaryReader = new BinaryReader(memStream))
+            {
+                binaryReader.Close();
+                binaryReader.Close();
+                binaryReader.Close();
+            }
+        }
+
+        [Fact]
+        public void BinaryWriter_DisposeTests_Negative()
```

Comment:
A pattern we use elsewhere, e.g. in the FileSystem tests, is to have a base abstract class with all the [Fact]s/[Theory]s and virtual/abstract methods that get overridden by derived types that provide overrides with the specific implementation they want tested.  So for example, we could have these dispose tests on a base class with an abstract method for disposing, and then have two derived types that simply override that method, one calling Dispose and the other calling Close. cc: @ianhays